src/containers/AccessPointDetails/components/Firmware/index.js

```diff
@@ -131,16 +134,19 @@ const Firmware = ({
         buttonText="Confirm"
       />
       <Form {...pageLayout} form={form} onValuesChange={handleOnFormChange}>
-        <div className={styles.InlineEndDiv}>
-          <Button
-            className={styles.saveButton}
-            onClick={handleOnReboot}
-            name="reboot"
-            disabled={getRebootStatus()}
-          >
-            Reboot AP
-          </Button>
-        </div>
+        {roleIsWritable && (
+          <div className={styles.InlineEndDiv}>
+            <Button
```

Comment:
add Button to WritableInputs, just make a new HOC that show/hides for Components